### PR TITLE
Added a bunch of docstrings, mainly for the cog processing chain

### PR DESCRIPTION
**Pull request by lucfr**
be aware: contains one potentially breaking change (9f33a1068f9bd9818d4e2a6d1b89a36d2b50bc47)

### CODE REVIEW

File: src/routers/cog.py [hunk 1]
```diff
@@ -20,8 +20,18 @@
 
 @router.put("/datasets/{dataset_id}/build-cog")
 def create_cog(dataset_id: int, options: Optional[ProcessOptions], token: Annotated[str, Depends(oauth2_scheme)], background_tasks: BackgroundTasks):
-    """
-    
+    """FastAPI process chain to add a cog-calculation task to the processing queue, with monitoring and logging.
+    Verifies the access token, loads the dataset to calculate, creates a TaskPayload and returns the task to start 
+    calculating in the background.
```
Comment (mmaelicke) [inline src/routers/cog.py:25]: to be more precise: It adds the task to the background process of FastAPI. The task metadata is returned to inform the user on the frontend about the queue position and estimated wait time

**Review by mmaelicke — CHANGES_REQUESTED** [review]
Thanks a lot!. There is only a minor change request.

**Comment by lucfr** [comment]
@mmaelicke 1fd7cd814585bb18ffb014339d322e0aeb948384 hopefully fixes your comments

**Review by mmaelicke — APPROVED** [review]
Nice, Thanks a lot!

**Pull request merged**